File: scratch/make_zip.php
<?php
$zipFile = 'update_frontend_final.zip';
if (file_exists($zipFile)) {
    unlink($zipFile);
}

$zip = new ZipArchive();
if ($zip->open($zipFile, ZipArchive::CREATE | ZipArchive::OVERWRITE) !== TRUE) {
    die("Failed to create zip file");
}

function addFolderToZip($folder, &$zip) {
    if (!is_dir($folder)) return;
    $files = new RecursiveIteratorIterator(
        new RecursiveDirectoryIterator($folder),
        RecursiveIteratorIterator::LEAVES_ONLY
    );
    foreach ($files as $name => $file) {
        if (!$file->isDir()) {
            $filePath = $file->getRealPath();
            $relativePath = substr($filePath, strlen(__DIR__) + 1);
            // Replace backslashes with forward slashes for zip internal paths
            $relativePath = str_replace('\\', '/', $relativePath);
            $zip->addFile($filePath, $relativePath);
        }
    }
}

addFolderToZip(__DIR__ . '/app', $zip);
addFolderToZip(__DIR__ . '/resources/views', $zip);
addFolderToZip(__DIR__ . '/public/build', $zip);

$zip->addFile(__DIR__ . '/routes/web.php', 'routes/web.php');
$zip->addFile(__DIR__ . '/clear_cache.php', 'clear_cache.php');
$zip->addFile(__DIR__ . '/deploy_frontend.php', 'deploy_frontend.php');

$zip->close();
echo "Zip created successfully. Structure preserved.\n";

File: scratch/make_zip.py
import os
import zipfile

def zipdir(path, ziph, root_path):
    if not os.path.exists(path):
        return
    for root, dirs, files in os.walk(path):
        for file in files:
            file_path = os.path.join(root, file)
            ziph.write(file_path, os.path.relpath(file_path, root_path))

with zipfile.ZipFile('update_frontend_final.zip', 'w', zipfile.ZIP_DEFLATED) as zipf:
    zipdir('app', zipf, '.')
    zipdir('resources/views', zipf, '.')
    zipdir('public/build', zipf, '.')
    zipdir('storage/app/public', zipf, '.') # Include images!
    zipf.write('routes/web.php', 'routes/web.php')
    zipf.write('clear_cache.php', 'clear_cache.php')
    zipf.write('deploy_frontend.php', 'deploy_frontend.php')

print("Zip created successfully with images.")
